site/src/components/DurationField/DurationField.stories.tsx

```diff
@@ -0,0 +1,50 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import { useState } from "react";
+import { DurationField } from "./DurationField";
+
+const meta: Meta<typeof DurationField> = {
+  title: "components/DurationField",
+  component: DurationField,
+  args: {
+    label: "Duration",
+  },
+  render: function RenderComponent(args) {
+    const [value, setValue] = useState<number | undefined>(args.value);
+    return (
+      <DurationField
+        {...args}
+        value={value}
+        onChange={(value) => setValue(value)}
+      />
+    );
+  },
```

Comment:
Noticed one other thing about this render function: what is `useState` doing in this case?

I'm not sure what causes the `render` method to reevaluate itself and rebuild the component from scratch, but from this code, I would expect `args.value` to be locked in during the initial render, and then anytime `args` changes, the new value would be ignored

I was having trouble changing the `value` prop, and I'm not sure if that's connected

Reply:
Since this component is a controlled input, we need some wrapper around it to change the value while typing.

Reply:
Marking as resolved because this should not happen anymore.

Reply:
Yeah, I get why we need some kind of state. I was just wondering if it might be worth making a new hook called `useWeakState` (or even `useStorybookState`), where it mostly behaves like `useState`, but if the value passed into the hook changes, that automatically throws out the current state value, and uses the new value as the fresh state

That way, you can have the controlled input from the wrapper component, but if Storybook pushes down a new value by calling the function with something else, the state will reset itself, basically. I know we have one other Storybook test that has `useState` set up like this